src/components/ContactImportation/ContactImportationModal.jsx

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  Alerter,
+  Button,
+  Modal,
+  ModalContent,
+  ModalHeader
+} from "cozy-ui/react";
+import _ from "lodash";
+import { withContactsMutations } from "../../connections/allContacts";
+import ContactImportationCard from "./ContactImportationCard";
+import Importation from "../../importation";
+import Status from "../../importation/status";
+
+const ERROR_STATUS_SET = new Set([Status.FILE_ISSUE, Status.COMPLETE_FAILURE]);
+
+class ContactImportationModal extends React.Component {
+  state = {
+    progress: null
+  };
+
+  selectFile = file => {
+    this.props.updateImportation(
+      Importation.selectFile(file, this.props.importation)
+    );
+  };
+
+  unselectFile = () => {
+    this.props.updateImportation(
+      Importation.unselectFile(this.props.importation)
+    );
+  };
+
+  onProgress = progress => {
+    this.setState({ progress });
+  };
+
+  importFile = async () => {
+    const { createContact, importation, updateImportation } = this.props;
+    const { runningImportation, finishedImportationPromise } = Importation.run(
+      importation,
+      {
+        save: createContact,
+        onProgress: this.onProgress
+      }
+    );
+
+    updateImportation(runningImportation);
+
+    const finishedImportation = await finishedImportationPromise;
+
+    if (finishedImportation.status === Status.COMPLETE_SUCCESS) {
+      Alerter.success("importation.complete_success", {
+        smart_count: _.get(finishedImportation, "report.total", "")
+      });
+      this.forgetImportation();
+    } else {
+      updateImportation(finishedImportation);
+    }
+  };
+
+  forgetImportation = () => {
+    this.props.updateImportation(null);
+  };
+
+  render() {
+    const { importation } = this.props;
+    const { t } = this.context;
+    const { status } = importation;
+
+    return (
+      <Modal
+        into="body"
+        size="small"
+        className={ERROR_STATUS_SET.has(status) && "importation-error"}
+        dismissAction={this.forgetImportation}
+      >
+        <ModalHeader title={t("importation.title")} />
+        <ModalContent>
```

Comment:
The `<ModalConte>` tree must be divided into smaller component like in [the paragraph "Break The UI Into A Component Hierarchy" in thinking in react](https://reactjs.org/docs/thinking-in-react.html#step-1-break-the-ui-into-a-component-hierarchy)